商品編集機能の確認

akashiyaki0128 opened the pull request:
What
編集機能の実装を行いました。すいません、マスターでコミット後にブランチしていないことに気づいたので、
空コミット後になります。以下実施内容ですので、確認お願い致します。
１，ルーティング
２，コントローラーでedit,update定義
３，編集画面へのリンク先を指定
４，editビューの編集　newと同一のものを使用
５，edit,updateの定義内記述updateの取得するデータの代入先は@itemにする、理由：エラーメッセージのフォームの求めるモデルを@itemで一致させる為
６，アクセス制限、プライベートメソッドにensure_current_userを定義し、before_actionでメソッド実行。if文はcurrent_userと@item.userのidを取得して比較するイメージで記述
７，rubocop実行済

編集画面に商品情報が残っている動画
https://i.gyazo.com/c1fcf680c040dc55451ec19968689ab4.mp4
編集失敗時の動画
https://i.gyazo.com/5cae82c0ac5a3ed30bb3856ba76727ae.mp4
編集成功時の動画
https://i.gyazo.com/f35afaad3876e387eafd7e6f75fb34bf.mp4
ログアウト時の編集ページへのアクセス制限
https://i.gyazo.com/00959e511ea2895dde2ecb5ed258a1fa.mp4
投稿ユーザー以外の編集ページへのアクセス制限
https://i.gyazo.com/3ea359540da01531821f7b508e25b56a.mp4

why
編集を可能にするため

akashiyaki0128 committed:
商品編集機能の確認

s-aka-moto commented on app/controllers/items_controller.rb:
```diff
@@ -36,7 +36,6 @@ def show
-  
   private
 
   def item_params
```
`@user = @item.user_id`
こちら@userを定義せず、
`if current_user.id != @item.user_id`としてはいかがでしょうか？

s-aka-moto commented on app/controllers/items_controller.rb:
```diff
@@ -36,7 +36,6 @@ def show
 
-  
   private
 
```
ensure_current_user内で@itemが定義されているため、edit、updateに@itemを定義する必要は無いかと存じます！
(updateにもensure_current_userを適用すると良いですね！)

akashiyaki0128 replied:
レビューありがとうございます！修正致しまた。
とても勉強になりました！！
１，ensure_current_user内の記述を変更
２，edit,updateともにbefore_actionで@item = Item.find(params[:id])を定義しているため同内容の記述を削除

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

akashiyaki0128 committed:
コードの修正

１，ensure_current_user内の記述を変更
２，edit,updateともにbefore_actionで@item = Itemf.find(params[:id])を定義しているため同内容の記述を削除

i-date reviewed (approved):
修正ありがとうございます！
修正点を拝見したところ問題ないことを確認できましたので、LGTMとさせていただきます。
以降の実装も引き続きこの調子で進めていきましょう。応援しております！
